activerecord/lib/active_record/connection_adapters/postgresql/result.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module ActiveRecord::ConnectionAdapters::PostgreSQL
+  ###
+  # This class encapsulates a result returned from calling
+  # {#exec_query}[rdoc-ref:ConnectionAdapters::DatabaseStatements#exec_query]
+  # on the PostgreSQL connection adapter.
+  class Result < ActiveRecord::Result
+    include Enumerable
+
+    attr_reader :columns
+
+    def initialize(pg_result, adapter)
+      @pg_result = pg_result
+      @adapter = adapter
+      @field_map = nil
+      @length = nil
+      @columns = @pg_result.fields
+    end
+
+    def initialize_copy(other)
```

Comment:
hmm what is going on with this blank method?

Reply:
This is called from the [query cache](https://github.com/rails/rails/blob/e0874b6f88509270d7603184ecd158d55df7204f/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L119). I'm uncertain why some of the data (not a deep copy of all the values) is duped [in AR::Result](https://github.com/rails/rails/blob/1cec4e1bbaba786aa4ea70a0e2b6ad6f15ec1e68/activerecord/lib/active_record/result.rb#L122-L127). They are treated as read-only, so that there should be no need to dup them. Removing `initialize_copy` from AR::Result doesn't raise any failure in `rake test:postgresql` or `rake test:sqlite3`. I only added this method because I derived the class from AR::Result to satisfy [some class based tests](https://github.com/rails/rails/blob/544492d04f10beb007df0db968c24bed140a1c4f/activerecord/test/cases/adapter_test.rb#L220), but the base method isn't suitable for PostgreSQL::Result.

We could dup ruby objects here as well as kind of precaution. At least PG::Result is immutable, so that it doesn't need to be copied.

Reply:
hmmm @tenderlove does it make sense to break off from this interface in Rails 6.0 and remove this. It is very odd. ... @larskanis  at a minimum we should add a comment explaining why it is no-op.